packages/react/src/components/Menu/MenuItem.tsx

```diff
@@ -97,26 +102,40 @@ export const MenuItem = forwardRef<HTMLLIElement, MenuItemProps>(
     },
     forwardRef
   ) {
+    const [submenuOpen, setSubmenuOpen] = useState(false);
+    const [rtl, setRtl] = useState(false);
+
+    const {
+      refs,
+      floatingStyles,
+      context: floatingContext,
+    } = useFloating({
+      open: submenuOpen,
+      onOpenChange: setSubmenuOpen,
+      placement: rtl ? 'left-start' : 'right-start',
+      whileElementsMounted: autoUpdate,
+      middleware: [offset({ mainAxis: -6, crossAxis: -6 })],
+    });
+    const { getReferenceProps, getFloatingProps } = useInteractions([
+      useHover(floatingContext, {
+        enabled: true,
+        handleClose: safePolygon({
+          requireIntent: true,
+        }),
```

Comment:
Here I'm trying to get to `Italic`, but can't get there even though I'm in the triangle. The `Share with` item is being invoked when I expect it to be ignored. Maybe the [`blockPointerEvents` option](https://floating-ui.com/docs/usehover#blockpointerevents) could help?


https://github.com/user-attachments/assets/7aebcc5a-a9ce-41e6-b94d-f3c2058f0dd1

Reply:
[blockPointerEvents](https://floating-ui.com/docs/usehover#blockpointerevents) was blocking the entire menu from any event.  its happening only when any other pointer comes under the safePolygon. If you try it with` Share with ` It works fine. So I added a `delay` of 100ms to avoid the issue.